### PR TITLE
LNK-4621: data acquisition query plan rest api not properly validating values

### PR DESCRIPTION
**Pull request by edward-miller-lcg**
### 🛠️ Description of Changes
Add enhanced validation to Query Plan when add/updating it.

### 🧪 Testing Performed
Local: 1. e2e, 2. validated use cases defined in ticket.

image

image

### 🧑‍🔬 Unit Testing
- [x] I have written or updated unit tests to cover my changes

### 📓 Documentation Updated
n.a

## Summary by CodeRabbit

* **New Features**
  * Introduced comprehensive validation for query plan configurations with enhanced error detection and improved error messaging.
  * Added automatic default values for paging parameters to ensure data consistency.

* **Tests**
  * Added extensive test coverage for query plan validation scenarios and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs [hunk 1]
```diff
@@ -37,9 +40,25 @@
...
-        ValidateQueryOrder(model.InitialQueries, "InitialQueries");
-        ValidateQueryOrder(model.SupplementalQueries, "SupplementalQueries");
+        // Perform comprehensive validation
+        var validationResult = _validator.ValidateQueryPlan(model.InitialQueries, model.SupplementalQueries);
+
+        if (!validationResult.IsValid)
+        {
+            _logger.LogError("Query Plan validation failed for facility {FacilityId}: {Errors}",
+                model.FacilityId,
+                string.Join("; ", validationResult.Errors));
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs]: complete

File: DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs [hunk 2]
```diff
@@ -37,9 +40,25 @@
...
-        //// Validate query order
-        ValidateQueryOrder(model.InitialQueries, "InitialQueries");
-        ValidateQueryOrder(model.SupplementalQueries, "SupplementalQueries");
+        // Perform comprehensive validation
+        var validationResult = _validator.ValidateQueryPlan(model.InitialQueries, model.SupplementalQueries);
+
+        if (!validationResult.IsValid)
+        {
+            _logger.LogError("Query Plan validation failed for facility {FacilityId}: {Errors}",
+                model.FacilityId,
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs]: complete

File: DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs [hunk 3]
```diff
@@ -37,9 +40,25 @@
...
+                string.Join("; ", validationResult.Errors));
+
+            throw new BadRequestException($"Query Plan validation failed: {validationResult.GetErrorMessage()}");
+        }
+
+        // Log warnings if any exist
+        if (validationResult.Warnings.Any())
+        {
+            _logger.LogWarning("Query Plan validation warnings for facility {FacilityId}: {Warnings}",
+                model.FacilityId,
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs]: complete.

File: DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs [hunk 4]
```diff
@@ -59,6 +78,10 @@
         entity = await _database.QueryPlanRepository.AddAsync(entity);
         await _database.QueryPlanRepository.SaveChangesAsync();
 
+        _logger.LogInformation("Successfully created Query Plan for facility {FacilityId} with type {Type}",
+            model.FacilityId,
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs]: complete.

File: DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs [hunk 5]
```diff
@@ -69,82 +92,88 @@
...
-        ValidateQueryOrder(model.SupplementalQueries, "SupplementalQueries");
+        // Perform comprehensive validation
+        var validationResult = _validator.ValidateQueryPlan(model.InitialQueries, model.SupplementalQueries);
 
-        var existingQueryPlan = await _database.QueryPlanRepository.FirstOrDefaultAsync(q => q.FacilityId == model.FacilityId && q.Type == model.Type);
+        if (!validationResult.IsValid)
+        {
+            _logger.LogError("Query Plan validation failed for facility {FacilityId}: {Errors}",
+                model.FacilityId,
+                string.Join("; ", validationResult.Errors));
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs]: complete

File: DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs [hunk 6]
```diff
@@ -69,82 +92,88 @@
...
         {
-            existingQueryPlan.InitialQueries = model.InitialQueries;
-            existingQueryPlan.SupplementalQueries = model.SupplementalQueries;
-            existingQueryPlan.PlanName = model.PlanName;
-            existingQueryPlan.EHRDescription = model.EHRDescription;
-            existingQueryPlan.LookBack = model.LookBack;
-            existingQueryPlan.ModifyDate = DateTime.UtcNow;
+            _logger.LogWarning("Query Plan validation warnings for facility {FacilityId}: {Warnings}",
+                model.FacilityId,
+                string.Join("; ", validationResult.Warnings));
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs]: complete

File: DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs [hunk 7]
```diff
@@ -69,82 +92,88 @@
...
+        existingQueryPlan.SupplementalQueries = model.SupplementalQueries;
+        existingQueryPlan.PlanName = model.PlanName;
+        existingQueryPlan.EHRDescription = model.EHRDescription;
+        existingQueryPlan.LookBack = model.LookBack;
+        existingQueryPlan.ModifyDate = DateTime.UtcNow;
+
+        await _database.QueryPlanRepository.SaveChangesAsync();
+
+        _logger.LogInformation("Successfully updated Query Plan for facility {FacilityId} with type {Type}",
+            model.FacilityId,
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs]: redundant

File: DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs [hunk 8]
```diff
@@ -37,9 +40,25 @@
...
+
+            throw new BadRequestException($"Query Plan validation failed: {validationResult.GetErrorMessage()}");
+        }
+
+        // Log warnings if any exist
+        if (validationResult.Warnings.Any())
+        {
+            _logger.LogWarning("Query Plan validation warnings for facility {FacilityId}: {Warnings}",
+                model.FacilityId,
+                string.Join("; ", validationResult.Warnings));
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs]: redundant

File: DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs [hunk 9]
```diff
@@ -69,82 +92,88 @@
...
-        ValidateQueryOrder(model.InitialQueries, "InitialQueries");
-        ValidateQueryOrder(model.SupplementalQueries, "SupplementalQueries");
+        // Perform comprehensive validation
+        var validationResult = _validator.ValidateQueryPlan(model.InitialQueries, model.SupplementalQueries);
 
-        var existingQueryPlan = await _database.QueryPlanRepository.FirstOrDefaultAsync(q => q.FacilityId == model.FacilityId && q.Type == model.Type);
+        if (!validationResult.IsValid)
+        {
+            _logger.LogError("Query Plan validation failed for facility {FacilityId}: {Errors}",
+                model.FacilityId,
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs]: redundant

File: DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs [hunk 10]
```diff
@@ -69,82 +92,88 @@
...
+        if (validationResult.Warnings.Any())
         {
-            existingQueryPlan.InitialQueries = model.InitialQueries;
-            existingQueryPlan.SupplementalQueries = model.SupplementalQueries;
-            existingQueryPlan.PlanName = model.PlanName;
-            existingQueryPlan.EHRDescription = model.EHRDescription;
-            existingQueryPlan.LookBack = model.LookBack;
-            existingQueryPlan.ModifyDate = DateTime.UtcNow;
+            _logger.LogWarning("Query Plan validation warnings for facility {FacilityId}: {Warnings}",
+                model.FacilityId,
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Managers/QueryPlanManager.cs]: redundant

File: DotNet/DataAcquisition.Domain/Application/Validators/QueryPlanValidator.cs [hunk 11]
```diff
@@ -0,0 +1,539 @@
...
+                result.Errors.Add($"{prefix}: Paged value '{parameter.Paged}' is not a valid int.");
+            }
+            else if (pagedVal < 0)
+            {
+                result.IsValid = false;
+                result.Errors.Add($"{prefix}: Paged value cannot be negative.");
+            }
+        }
+
+    }
```
[the remark this answers is not in the document]
Comment (edward-miller-lcg) [reply DotNet/DataAcquisition.Domain/Application/Validators/QueryPlanValidator.cs:363]: resolved

**Review by dvargaslantana — APPROVED** [review]
